docs/rst/manual/developer_documentation/storage_server.rst

```diff
@@ -0,0 +1,172 @@
+ERT Storage Server
+==================
+
+The ERT Storage Server API has two implementations. The PostgreSQL (and
+optionally Azure Blob Storage) centralised server is implemented by `ERT Storage
+Server <https://github.com/equinor/ert-storage>`_ while the block-storage-based
+is implemented in the ERT repository by the "Dark Storage".
+
+.. uml:: storage_server_er_diagram.plantuml
+
+*Entities highlighted in green represent concepts that are part of the public
+API. That is, they're not just implementation details but concepts that we
+expose to the user.*
+
+`Experiment`_
+    A specific ERT configuration, reservoir model and observations.
+
+`Ensemble`_
+    A single ERT ensemble evaluation.
+
+`Observation`_
+    A named group of observations.
+
+    Indirectly connected to 'response' records by name.
+
+Prior
+    The statistical distribution used to generate the parameters of a given
+    ensemble. The ``function`` field is really an enum where each
+
+ObservationTransformation
+    ERT might have a workflow that scales observations or deactivates them
+    between ensemble evaluations in an iterative ensemble smoother run.
```

Comment:
Don't think we should specifically mention "iterative ensemble smoother" here. Perhaps just say between ensemble evaluations in an update step or something like that?